src/main/web/components/forms/inputs/DropdownFormSelector.tsx

```diff
@@ -0,0 +1,337 @@
+            const errors = FieldValue.getErrors(nonEmpty).push({
+              kind: ErrorKind.Loading,
+              message: `Failed to load value set`,
+            });
+            return FieldValue.setErrors(nonEmpty, errors);
+          });
+        },
+      });
+    } else {
+      this.isLoading = false;
+    }
+  }
+
+  componentDidMount() {
+    this.initValueSet()
+    // tryExtractNestedForm(this.props.children, this.context, this.props.nestedFormTemplate)
+    //   .then(nestedForm => {
+    //     if (nestedForm != undefined) {
+    //       this.setState({nestedForm});
+    //     }
+    //   });
+  }
+
+  componentWillUnmount() {
+    this.cancellation.cancelAll();
+  }
+
+  private onValueChanged = (value?: SparqlBindingValue) => {
+    this.setState({ nestedFormOpen: false });
+    this.setAndValidate(this.parseValue(value));
+  };
+
+  private toggleNestedForm = () => {
+    this.setState((state): State => ({ nestedFormOpen: !state.nestedFormOpen }));
+  };
+
+  private parseValue(value: SparqlBindingValue): AtomicValue | EmptyValue {
+    // this is for testing purpose only i.e. checking whether callback is called
+    if (!value) {
+      return FieldValue.empty;
+    }
+
+    const findCorresponding = this.state.valueSet.find((v) => v.value.equals(value.value));
+    if (!findCorresponding) {
+      return FieldValue.empty;
+    }
+
+    // turn into field value for standard validation calls
+    const corresponding: Partial<AtomicValue> = {
+      value: findCorresponding.value,
+      label: findCorresponding.label,
+    };
+    return AtomicValue.set(this.props.value, corresponding);
+  }
+
+  private optionRenderer = (option: SparqlBindingValue) => {
+    if (this.props.template !== undefined) {
+      return createElement(TemplateItem, {
+        template: {
+          source: this.props.template,
+          options: option.binding,
+        },
+      });
+    } else {
+      // default option template
+      return D.span({ id: option.label, className: OPTION_CLASS }, option.label || option.value.value);
+    }
+  };
+
+  private valueRenderer = (v: AtomicValue | undefined) => {
+    // that is if user adds a new input which get's empty as initial field value
+    if (!v) {
+      return;
+    }
+
+    const valueSet = this.state.valueSet;
+    if (valueSet) {
+      // try to find the selected value in the pre-computed valueSet
+      const bindingValue = valueSet.find((setValue) => setValue.value.equals(v.value));
+      // if existing, then use optionRenderer to exploit the template and additional bindings
+      if (bindingValue) {
+        return this.optionRenderer(bindingValue);
+      }
+    }
+
+    // fallback rendering i.e. if recovering from state or saved value
+    // but value is not any longer in dynamically (on every initialization) computed set
+    return D.span({ id: v.label, className: OPTION_CLASS }, v.label || v.value.value);
+  };
+
+  private openSelectedNestedForm(label: string) {
```

Comment:
when user select a label from dropdown, the related nested form is stored in the component state and the nested form is opened